core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiServiceSpec.scala

```diff
@@ -986,4 +986,144 @@ class SubmissionApiServiceSpec extends ApiServiceSpec with TableDrivenPropertyCh
+      JsString("This comment has 1000 characters - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean tristique leo purus, non rutrum arcu posuere in. Quisque at venenatis augue. Nullam egestas dapibus lacus, pretium pretium neque semper quis. Cras dictum orci id vestibulum molestie. Duis tincidunt magna et turpis rutrum, in sagittis nibh ornare. " +
+        "Aliquam facilisis interdum diam, eget tempus dolor. Vivamus porta, enim in interdum dictum, arcu nulla maximus nulla, ac lacinia justo quam vel lectus. Donec a nunc in justo vulputate dignissim sit amet ut mi. Donec pulvinar, enim ut bibendum suscipit, leo urna dignissim turpis, at sagittis lorem leo vel ante. Suspendisse viverra quam vel imperdiet " +
+        "congue. Maecenas vitae viverra dolor, semper congue nibh. Duis nec ante facilisis dui tempor fringilla quis sed erat. Praesent eget ipsum id felis accumsan posuere non a magna. Cras risus velit, facilisis in erat eget, auctor malesuada tellus. Proin posuere risus non lectus efficitur consequat. Sed ut."))
+  )
+
+  forAll(validUserCommentCases) {
+    (description, userCommentInput, userCommentResult) =>
+      it should description in {
+        withTestDataApiServices { services =>
+          val workspaceName = testData.wsName
+          val methodConfigurationName = MethodConfigurationName("no_input", "dsde", workspaceName)
+          ensureMethodConfigs(services, workspaceName, methodConfigurationName)
+
+          Post(
+            s"${workspaceName.path}/submissions",
+            JsObject(
+              requiredSubmissionFields(methodConfigurationName, testData.sample1) ++
+                List(
+                  userCommentInput.map(x => "userComment" -> x.toJson),
+                ).flatten: _*
+            )
+          ) ~>
+            sealRoute(services.submissionRoutes) ~>
+            check {
+              val response = responseAs[String]
+              status should be(StatusCodes.Created)
+              val requestUserComment = getResponseField(response, "userComment")
+              requestUserComment.get shouldBe userCommentResult
+            }
+        }
+      }
+  }
+
+  it should "return a parameter error if the userComment is invalid" in {
+    withTestDataApiServices { services =>
+      val workspaceName = testData.wsName
+      val methodConfigurationName = MethodConfigurationName("no_input", "dsde", workspaceName)
+      ensureMethodConfigs(services, workspaceName, methodConfigurationName)
+
+      val invalidUserComment =
+        """
+          |This comment is invalid since it has more than 1000 characters - Lorem ipsum dolor sit amet, consectetur adipiscing elit.
```

Comment:
Does Scala not have a way to repeat a character a certain number of times (there is a simple syntax for this in Python)? If it does, that would simplify the test case and make it obvious how many characters are represented.

Reply:
TIL that there is a class called `RandomStringUtils` to randomly generate string of desired length. I will update the code to use that function. Thank you!